### ISSUE TITLE
Why do I want the 315k network file?

### ISSUE THREAD

**Issue by Marcin1960** [gcp/leela-zero#184]
I have an idea I would like to test.

BOTH Leela Zero and AlphaGo Zero develop with ups and downs. Could it be that it is a part of natural
learning process? I mean downs take place when the system absorbs new techniques - it slows  down for a time. If so the later "weaker" network file might be actually better!

I would like to play against 315K to check visually if there are new ways learned in comparison with earlier network.

**Comment by Marcin1960** [comment]
If mine wild guess is right (I did not get 315K file yet to verify it), one should stick to this network file until strength improves, otherwise some learning might be lost and the progress will be delayed.

Let me repeat, the same zig-zag pattern took place in AlphaGo Zero and it might be normal and desirable.

**Comment by axaj** [comment]
Yes, these fluctuations in strength are expected to happen.

I don't think the learning works the way you think it does. Nothing is lost if we don't use a trained network that is worse than the previous one. Every network gets trained from scratch. It's just the input data (the self-play games) that changes as more and more games are being generated by the current strongest network.

**Comment by Marcin1960** [comment]
Yes, but my point is that 315K file might an improvement, to be continued. Can someone give me the  315K file?

**Comment by OmnipotentEntity** [comment]
According to the Alpha Go Zero paper:

> AlphaGo Zero's self­play training pipeline consists of three main components, all executed asynchronously in parallel. Neural network parameters θ\_i are continually optimized from recent self­play data; AlphaGo Zero players α\_{θ\_i} are continually evaluated; and the best performing player so far, α\_{θ^∗}, is used to generate new self­play data.

Which is why we are using the network with the highest ELO.

For what it's worth, because the training data lookback is 500k games, we're not "losing" anything by not using the weaker engine.  Any hard-won insight is still present in the training data.

**Comment by Marcin1960** [comment]
" Any hard-won insight is still present in the training data."

Isn't there some element of randomness?

**Comment by OmnipotentEntity** [comment]
Again, we're doing the same thing as Alpha Go Zero did.  It's hard to reason about what's exactly going on inside of a NN, so it's dangerous to categorically say that the weakness is due to absorption of new insights.

Mathematically, what's going on is the gradient of a cost function with thousands of inputs is being found and then we walk a small amount in that direction.  We could easily have gone a bit too far in any one direction which causes us to go back up the cost function instead of down.  There's nothing that says the new NN must be, or is even likely to be, conceptually better, but just happens to be poorer at execution.

**Comment by Marcin1960** [comment]
" it's dangerous to categorically say that the weakness is due to absorption of new insights."

That is why I would like this 315K file, so I could have a look at the game and report what I saw. I guess I will not get anything :(

**Comment by OmnipotentEntity** [comment]
I don't control the data.  @gcp does. However, if you wish you can reconstruct the data yourself from the published sgf files available from the website as soon as they are updated.  All of the code necessary is in the training subfolder.

**Comment by Marcin1960** [comment]
"you can reconstruct the data yourself from the published sgf files available from the website as soon as they are updated"

Couple of sgf files would be enough. I just want to look at the game. Oh, well.

**Comment by axaj** [comment]
I guess it wouldn't hurt to upload the sgf files of the games used to evaluate a new network.

**Comment by Marcin1960** [comment]
It would not, but I do not hold my breath.

**Comment by imsqueekyclean** [comment]
So you are going to evaluate the network by looking at a few games? I don't understand what's going on here. How would that be statistically significant and how would you determine what was random play and intent/improvement?

**Comment by Marcin1960** [comment]
I wanted to have a look at the games. Same way as one can watch a game between two humans and see their styles and differences.

**Comment by gcp** [comment]
I uploaded some of those networks to https://sjeng.org/zero

Games can be easily reconstructed with autogtp or any GTP interface.
